python/GafferSceneTest/PrimitiveVariableQueryTest.py

```diff
@@ -0,0 +1,656 @@
+#  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+#  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+#  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+#  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+#  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+#  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+#  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+##########################################################################
+
+import unittest
+import imath
+
+import IECore
+import IECoreScene
+import Gaffer
+import GafferScene
+import GafferSceneTest
+
+class PrimitiveVariableQueryTest( GafferSceneTest.SceneTestCase ):
+
+	nonVectorSuffix = "NonVector"
+
+	names = (
+		"b",
+		"i",
+		"f",
+		"s",
+		"v2i",
+		"v3i",
+		"v2f",
+		"v3f",
+		"c3f",
+	)
+
+	dataTypes = (
+		IECore.BoolData,
+		IECore.IntData,
+		IECore.FloatData,
+		IECore.StringData,
+		IECore.V2iData,
+		IECore.V3iData,
+		IECore.V2fData,
+		IECore.V3fData,
+		IECore.Color3fData,
+	)
+
+	vectorDataTypes = (
+		IECore.BoolVectorData,
+		IECore.IntVectorData,
+		IECore.FloatVectorData,
+		IECore.StringVectorData,
+		IECore.V2iVectorData,
+		IECore.V3iVectorData,
+		IECore.V2fVectorData,
+		IECore.V3fVectorData,
+		IECore.Color3fVectorData,
+	)
+
+	values = (
+		True,
+		10,
+		11.0,
+		"foo",
+		imath.V2i( 12, 22 ),
+		imath.V3i( 13, 23, 33 ),
+		imath.V2f( 120.0, 220.0 ),
+		imath.V3f( 130.0, 230.0, 330.0 ),
+		imath.Color3f( 250.0, 450.0, 650.0 ),
+	)
+
+	defaultValues = (
+		False,
+		2,
+		6.0,
+		"baa",
+		imath.V2i( 62, 72 ),
+		imath.V3i( 63, 73, 73 ),
+		imath.V2f( 620.0, 720.0 ),
+		imath.V3f( 630.0, 730.0, 830.0 ),
+		imath.Color3f( 150.0, 350.0, 550.0 ),
+	)
+
+	plugTypes = (
+		Gaffer.BoolPlug,
+		Gaffer.IntPlug,
+		Gaffer.FloatPlug,
+		Gaffer.StringPlug,
+		Gaffer.V2iPlug,
+		Gaffer.V3iPlug,
+		Gaffer.V2fPlug,
+		Gaffer.V3fPlug,
+		Gaffer.Color3fPlug,
+	)
+
+	vectorPlugTypes = (
+		Gaffer.BoolVectorDataPlug,
+		Gaffer.IntVectorDataPlug,
+		Gaffer.FloatVectorDataPlug,
+		Gaffer.StringVectorDataPlug,
+		Gaffer.V2iVectorDataPlug,
+		Gaffer.V3iVectorDataPlug,
+		Gaffer.V2fVectorDataPlug,
+		Gaffer.V3fVectorDataPlug,
+		Gaffer.Color3fVectorDataPlug,
+	)
+
+	interpolations = (
+		IECoreScene.PrimitiveVariable.Interpolation.Constant,
+		IECoreScene.PrimitiveVariable.Interpolation.Uniform,
+		IECoreScene.PrimitiveVariable.Interpolation.Vertex,
+		IECoreScene.PrimitiveVariable.Interpolation.Varying,
+		IECoreScene.PrimitiveVariable.Interpolation.FaceVarying,
+	)
+
+	def makeQuad( self ) :
+
+		verticesPerFace = IECore.IntVectorData( [4] )
+		vertexIds = IECore.IntVectorData( [0, 1, 2, 3] )
+		p = IECore.V3fVectorData( [imath.V3f( 0, 0, 0 ), imath.V3f( 1, 0, 0 ), imath.V3f( 1, 1, 0 ), imath.V3f( 0, 1, 0 )] )
+		mesh = IECoreScene.MeshPrimitive( verticesPerFace, vertexIds, "linear", p )
```

Comment:
I think this could just be a single call to `MeshPrimitive.createPlane()`.

Reply:
done